generator/karaf/src/main/java/io/fabric8/maven/generator/karaf/KarafGenerator.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.generator.karaf;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.fabric8.maven.core.util.Configs;
+import io.fabric8.maven.core.util.MavenUtil;
+import io.fabric8.maven.docker.config.AssemblyConfiguration;
+import io.fabric8.maven.docker.config.BuildImageConfiguration;
+import io.fabric8.maven.docker.config.ImageConfiguration;
+import io.fabric8.maven.generator.api.BaseGenerator;
+import io.fabric8.maven.generator.api.MavenGeneratorContext;
+import io.fabric8.utils.Strings;
+import org.apache.maven.project.MavenProject;
+
+public class KarafGenerator extends BaseGenerator {
+    public KarafGenerator(MavenGeneratorContext context) {
+        super(context, "karaf");
+    }
+
+    private enum Config implements Configs.Key {
+        combine        {{ d = "false"; }},
+        name           {{ d = "%g/%a:%l"; }},
+        alias          {{ d = "karaf"; }},
+        baseImage      {{ d = "fabric8/s2i-karaf:1.3"; }},
+        baseDir        {{ d = "/deployments/"; }},
+        user           {{ d = "jboss:jboss:jboss"; }},
+        cmd            {{ d = "/deployments/deploy-and-run.sh"; }},
+        webPort        {{ d = "8181"; }},
+        jolokiaPort    {{ d = "8778"; }},
+        prometheusPort {{ d = "9779"; }};
```

Comment:
AFAIK the s2i-karaf base images does not use agent-bond (and hence does not export Prometheus' jmx_exporter). Guess we can savely remove this here.